spring-integration-core/src/main/java/org/springframework/integration/gateway/GatewayProxyFactoryBean.java

```diff
@@ -821,7 +828,7 @@ protected void doStop() {
 		}
 	}
 
-	private static boolean hasReturnParameterizedWithMessage(Method method, boolean runningOnCallerThread) {
```

Comment:
Any reason why?

Reply:
Yeah... I wanted to move `isReturnTypeMessage` logic into the `MethodInvocationGateway`, but looks like `runningOnCallerThread` is an obstacle, so I've decided to go the lazy-load approach.
Will reinstate this `static` in the next commit.

Thanks for catching this!